src/poprox_storage/repositories/articles.py

```diff
@@ -170,8 +173,28 @@ def store_mention(self, mention: Mention) -> UUID | None:
     insert_entity = store_entity
     insert_mention = store_mention
 
-    def _get_articles(self, article_table, where_clause=None) -> list[Article]:
-        query = article_table.select()
+    def _get_articles(self, article_table: Table, where_clause=None) -> list[Article]:
+        # Select only the most recent article row for each source/external id pair
+        inner_query = (
```

Comment:
This might be pre-mature optimization -- but do we want to put the where clause into the inner query instead of leaving it in the outer query?

Reply:
I thought about that too! The thing that prevented me from doing so is that the inner query doesn't fetch all the fields so can't support arbitrary where clauses on the table. There may be a way to optimize this, but it wasn't immediately obvious to me how